### PR TITLE
Migration from gradle to maven

### PR DESCRIPTION
**Pull request by znvbnea**
Migration is completed from gradle to maven

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    <artifactId>eiffel-remrem-publish</artifactId>
+    <version>0.6.2</version>
+    <packaging>pom</packaging>
+    <properties>
+        <slf4j-api.version>1.7.25</slf4j-api.version>
+        <!-- <spring-boot-starter-web.version>1.4.1.RELEASE</spring-boot-starter-web.version> 
+            <spring-boot-starter-actuator.version>1.4.1.RELEASE</spring-boot-starter-actuator.version> 
+            <spring-boot-starter-security.version>1.4.1.RELEASE</spring-boot-starter-security.version> -->
+        <spring-security-ldap.version>4.1.3.RELEASE</spring-security-ldap.version>
+        <spring-boot.version>1.4.1.RELEASE</spring-boot.version>
```
Comment (xmairaa) [inline pom.xml]: Please use 2.0.1.RELEASE

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        <gson.version>2.8.1</gson.version>
+        <javax.inject.version>1</javax.inject.version>
```
Comment (xmairaa) [inline pom.xml]: this version need to change to according bazaar availability

File: publish-cli/.gitignore [hunk 3]
```diff
@@ -1 +1,2 @@
 /bin/
+/target/
```
Comment (xmairaa) [inline publish-cli/.gitignore]: add eclipse and InteliJ related ignore files

File: publish-cli/pom.xml [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+            <groupId>javax.inject</groupId>
+            <artifactId>javax.inject</artifactId>
+            <version>${javax.inject.version}</version>
+            <scope>compile</scope>
+            <exclusions>
+                <exclusion>
+                    <artifactId>logback-classic</artifactId>
+                    <groupId>ch.qos.logback</groupId>
+                </exclusion>
+            </exclusions>
```
Comment (xmairaa) [inline publish-cli/pom.xml]: why exclusion for each dependency level?

File: publish-cli/pom.xml [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+        </dependency>
+    </dependencies>
+    <build>
+        <plugins>
```
Comment (xmairaa) [inline publish-cli/pom.xml]: Question : do we need this?

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,95 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
+    xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>com.github.Ericsson</groupId>
+    <artifactId>eiffel-remrem-publish</artifactId>
+    <version>0.6.2</version>
```
Comment (xmairaa) [inline pom.xml]: comment about version uplift

File: publish-service/pom.xml [hunk 7]
```diff
@@ -0,0 +1,167 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
+    xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>com.github.Ericsson</groupId>
+        <artifactId>eiffel-remrem-publish</artifactId>
+        <version>0.6.2</version>
```
Comment (xmairaa) [inline publish-service/pom.xml]: Please use <global.version> property in the parent pom, so Child are also follow that same version.

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    <packaging>pom</packaging>
+    <properties>
+        <slf4j-api.version>1.7.25</slf4j-api.version>
+        <!-- <spring-boot-starter-web.version>1.4.1.RELEASE</spring-boot-starter-web.version> 
+            <spring-boot-starter-actuator.version>1.4.1.RELEASE</spring-boot-starter-actuator.version>
```
Comment (xmairaa) [inline pom.xml]: remove this commented lines

File: pom.xml [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+        <springfox-swagger-ui.version>2.6.1</springfox-swagger-ui.version>
+        <commons-lang3.version>3.7</commons-lang3.version>
+        <eiffel-remrem-shared.version>0.3.6</eiffel-remrem-shared.version>
+        <jasypt-spring-boot-starter.version>2.0.0</jasypt-spring-boot-starter.version>
+        <eiffel-remrem-semantics.version>0.4.4</eiffel-remrem-semantics.version>
+        <commons-cli.version>1.3.1</commons-cli.version>
+        <junit.version>4.12</junit.version>
+        <assertj-core.version>3.4.1</assertj-core.version>
+        <json-path-assert.version>2.4.0</json-path-assert.version>
+        <!-- <spring-boot-starter-test.version>1.4.1.RELEASE</spring-boot-starter-test.version> -->
```
Comment (xmairaa) [inline pom.xml]: remove this commented line

File: pom.xml [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+        <springfox-swagger2.version>2.6.1</springfox-swagger2.version>
+        <springfox-swagger-ui.version>2.6.1</springfox-swagger-ui.version>
+        <commons-lang3.version>3.7</commons-lang3.version>
+        <eiffel-remrem-shared.version>0.3.6</eiffel-remrem-shared.version>
+        <jasypt-spring-boot-starter.version>2.0.0</jasypt-spring-boot-starter.version>
+        <eiffel-remrem-semantics.version>0.4.4</eiffel-remrem-semantics.version>
+        <commons-cli.version>1.3.1</commons-cli.version>
+        <junit.version>4.12</junit.version>
+        <assertj-core.version>3.4.1</assertj-core.version>
+        <json-path-assert.version>2.4.0</json-path-assert.version>
```
Comment (xmairaa) [inline pom.xml]: follow asc order

File: pom.xml [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+        <!-- <spring-boot-starter-web.version>1.4.1.RELEASE</spring-boot-starter-web.version> 
+            <spring-boot-starter-actuator.version>1.4.1.RELEASE</spring-boot-starter-actuator.version> 
+            <spring-boot-starter-security.version>1.4.1.RELEASE</spring-boot-starter-security.version> -->
+        <spring-security-ldap.version>4.1.3.RELEASE</spring-security-ldap.version>
+        <spring-boot.version>1.4.1.RELEASE</spring-boot.version>
+        <gson.version>2.8.1</gson.version>
+        <javax.inject.version>1</javax.inject.version>
+        <springfox-swagger2.version>2.6.1</springfox-swagger2.version>
+        <springfox-swagger-ui.version>2.6.1</springfox-swagger-ui.version>
+        <commons-lang3.version>3.7</commons-lang3.version>
```
Comment (xmairaa) [inline pom.xml]: check this commons-lang3 usability in application, if no use remove

File: publish-service/pom.xml [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+            <version>${spring-boot.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.jayway.jsonpath</groupId>
```
Comment (xmairaa) [inline publish-service/pom.xml]: check this json-path-assert usability in the application, if no use remove

File: publish-cli/pom.xml (hunk 5 above)
Comment (znvbnea) [reply publish-cli/pom.xml]: yes we need this . In cli module sure fire reports are not generating that's why we added this plugin

File: publish-cli/pom.xml (hunk 4 above)
Comment (znvbnea) [reply publish-cli/pom.xml]: added where we need to exclusion this dependency remaining places removed.

File: pom.xml [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+		<module>publish-common</module>
+		<module>publish-service</module>
+		<module>publish-cli</module>
+	</modules>
+	<repositories>
+		<repository>
+			<id>jitpack.io</id>
+			<name>Jitpack Repository</name>
+			<url>https://jitpack.io/</url>
+		</repository>
```
Comment (xdurvak) [inline pom.xml]: add the below repository also 
maven{
        url 'https://repo.spring.io/libs-milestone'
 }

Comment (xdurvak) [reply pom.xml]: check xml formatting

File: pom.xml [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
+	xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+	<modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>com.github.Ericsson</groupId>
+        <artifactId>eiffel-remrem-parent</artifactId>
+        <version>0.0.2</version>
+    </parent>
+    <properties>
+    <eiffel-remrem-publish.version>0.6.2</eiffel-remrem-publish.version>
```
Comment (xdurvak) [inline pom.xml]: I guess <eiffel-remrem-publish.version>0.6.2</eiffel-remrem-publish.version> property declaration not needed , specify version  directly

File: pom.xml (hunk 13 above)
Comment (znvbnea) [reply pom.xml]: changes done

File: publish-cli/pom.xml [hunk 15]
```diff
@@ -0,0 +1,171 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
+    xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>com.github.Ericsson</groupId>
+        <artifactId>eiffel-remrem-publish</artifactId>
+        <version>0.6.2</version>
```
Comment (xdurvak) [inline publish-cli/pom.xml]: I guess for sub modules you can use ${project.version} try this once

File: publish-common/pom.xml [hunk 16]
```diff
@@ -0,0 +1,122 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
+    xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>com.github.Ericsson</groupId>
+        <artifactId>eiffel-remrem-publish</artifactId>
+        <version>0.6.2</version>
```
Comment (xdurvak) [inline publish-common/pom.xml]: here also use the same ${project.version}

File: publish-service/pom.xml [hunk 17]
```diff
@@ -0,0 +1,170 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd"
+    xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>com.github.Ericsson</groupId>
+        <artifactId>eiffel-remrem-publish</artifactId>
+        <version>0.6.2</version>
```
Comment (xdurvak) [inline publish-service/pom.xml]: here also use the same ${project.version}

**Comment by xmairaa** [comment]
I will create new a story for **evasiba-ericsson** comments. Merge This pull request.

File: pom.xml [hunk 18]
```diff
@@ -0,0 +1,89 @@
...
+            <name>Spring Repository</name>
+            <url>https://repo.spring.io/libs-milestone</url>
+        </repository>
+    </repositories>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>3.7.0</version>
```
Comment (xmairaa) [inline pom.xml]: use 3.6.1

File: publish-cli/pom.xml [hunk 19]
```diff
@@ -0,0 +1,171 @@
...
+                            <skip><!-- true or the result of a Maven/system 
+                                    property for example -->
+                            </skip>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
```
Comment (xmairaa) [inline publish-cli/pom.xml]: In Effective pom.xml its taking 2.12.4, so Please mention version to 2.22.0

File: publish-service/pom.xml [hunk 20]
```diff
@@ -0,0 +1,170 @@
...
+                </exclusion>
+            </exclusions>
+        </dependency>
+    </dependencies>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-war-plugin</artifactId>
+                <version>2.6</version>
```
Comment (xmairaa) [inline publish-service/pom.xml]: use 2.2 only

**Pull request merged**